applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -85,121 +86,113 @@ pub struct TransientData {
 }
 
 // setup curl authentication
-fn base_curl_auth(curl: &mut Easy) -> Result<(), MmProxyError> {
-    curl.username(MONEROD_USER)
-        .map_err(|_| MmProxyError::CurlError("Invalid username".to_string()))?;
-    curl.password(MONEROD_PASS)
-        .map_err(|_| MmProxyError::CurlError("Invalid password".to_string()))?;
+fn base_curl_auth(curl: &mut Easy) {
+    curl.username(MONEROD_USER).unwrap();
```

Comment:
Haven't managed to get it to panic on these unwraps, even with an invalid config.

The errors are presented on the perform() function, which will be caught by the match.

There was an unchecked unwrap on line 657 which is fixed now and appears to be the source of the poisoned lock error.

Reply:
Try to avoid unwraps in general. Even if they are 100% fine, it makes it harder and harder to review because you then have to check everywhere to see if the unwrap is ok

Reply:
Good point. Am updating the code.